core/libcamera_app.cpp

```diff
@@ -11,10 +11,38 @@
 #include "core/libcamera_app.hpp"
 #include "core/options.hpp"
 
+#include <fcntl.h>
+
+#include <sys/ioctl.h>
+
+#include <linux/videodev2.h>
+
+// If we definitely appear to be running the old camera stack, complain and give up.
+// Everything else, Pi or not, we let through.
+
+static void check_camera_stack()
+{
+	int fd = open("/dev/video0", O_RDWR, 0);
+	if (fd < 0)
+		return;
+
+	v4l2_capability caps;
+	int ret = ioctl(fd, VIDIOC_QUERYCAP, &caps);
+	close(fd);
+
+	if (ret < 0 || strcmp((char *)caps.driver, "bm2835 mmal"))
+		return;
+
+	fprintf(stderr, "ERROR: the system appears to be configured for the legacy camera stack\n");
+	exit(-1);
```

Comment:
Should we add instruction (or perhaps a url) to detail how to switch back to libcamera?

Reply:
Yes, might be a good idea - maybe a link to the part of the documentation page that describes what to do? Of course that would have to be written first... :)

Maybe merge this now, and we can change the message once the documentation has been taken care of?